modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1355,7 +1373,16 @@ public CompletableFuture<Void> onUpdate(WatchEvent evt) {
 
                     assert newDataNodes != null : "Data nodes was not initialized.";
 
-                    zoneState.nodes(newDataNodes);
+                    String filter;
+
+                    try {
+                        filter = getZoneById(zonesConfiguration, zoneId).filter().value();
+                    } catch (DistributionZoneNotFoundException ignored) {
```

Comment:
How it's possible to catch DistributionZoneNotFoundException here?

Reply:
`DistributionZonesUtil#getZoneById` throws  `DistributionZoneNotFoundException` if there is no such zone with `zoneId` in `zonesConfiguration`.
We could receive event about dataNodes changes for `zoneId`, and while we was handling this watch event, zone could be deleted

Reply:
What do you mean? I believe that zone won't be deleted from local configuration before we will retrieve corresponding notification from meta storage which isn't possible because we are in ms notification thread.

Reply:
Ok, I totally forgot about the fact, that parallel watches feature was reverted, you are right that my scenario is invalid, I've removed catch clause